### PR TITLE
Add optional parameter to REST API to allow filter by only live collexes

### PR DESCRIPTION
**Pull request by NickGrantONS**
# Motivation and Context
We are currently filtering in the front end, to display only the live collection exercises. By adding an optional parameter to the REST API, we can simplify the front-end code and reduce the amount of data being transferred from DB -> Collex Service -> Front-end.

# What has changed
Added optional param to `CollectionExerciseEndpoint.getCollectionExercisesForSurvey`.

# How to test?
There's a new unit/REST test which covers this change.

To manually test, send a GET to `http://localhost:8145/collectionexercises/survey/{id}?liveOnly=true` and you should only receive collection exercises which are in the LIVE state.

# Links
Trello: https://trello.com/c/n7EcEA9y/344-add-ability-to-collection-exercise-to-only-return-live-collection-exercises-for-a-survey-m

### CODE REVIEW

File: src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/impl/CollectionExerciseServiceImpl.java [hunk 1]
```diff
@@ -102,6 +103,12 @@ public CollectionExerciseServiceImpl(
     return this.collectRepo.findBySurveyId(UUID.fromString(survey.getId()));
   }
 
+  @Override
+  public List<CollectionExercise> findCollectionExercisesBySurveyIdAndState(
+      UUID surveyId, CollectionExerciseState state) {
+    return collectRepo.findBySurveyIdAndState(surveyId, state);
```
Comment (MIslam92) [inline src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/impl/CollectionExerciseServiceImpl.java]: Codecov is complaining this is not covered in a test might also be worth adding one for it

Comment (NickGrantONS) [reply src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/impl/CollectionExerciseServiceImpl.java]: a unit test for a 1-liner seems excessive

Comment (NickGrantONS) [reply src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/impl/CollectionExerciseServiceImpl.java]: we're going to refactor away the stupid wrapper functions around repo methods

Comment (MIslam92) [reply src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/impl/CollectionExerciseServiceImpl.java]: I thought bout that too but just codecov being codecov but yeah if we're removing them wrapper functions thats cool 👍

File: API.md [hunk 2]
```diff
@@ -18,6 +18,7 @@ This page documents the Collection Exercise service API endpoints. Apart from th
 
 ## Get Collection Exercises for Survey
 * `GET /collectionexercises/survey/{survey_id}` will return a list of known collection exercises for the survey id.
+* `GET /collectionexercises/survey/{survey_id}?liveOnly=true` will return a list of LIVE collection exercises for the survey id.
```
Comment (tej99) [inline API.md]: 👍

**Pull request merged**